### PR TITLE
Add documentation and code patches for maintenability

### PR DESCRIPTION
**Pull request by pierre-dejoue**
- [x] Add documentation to README
- [x] List headers in cmake files
- [x] Code patches that should not impact functionality. The goal is readability and maintenability

Note: I found Thomas's image on Flipscan followin that link: https://blackflux.wordpress.com/2014/03/01/meshing-in-voxel-engines-part-2/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,3 +73,34 @@ Examples:
...
 ```
+
+BUILD WITH CMAKE
+================
+
+Build the library
+-----------------
+
+```
+mkdir build && cd make
```
Comment (jhasse) [inline README.md:84]: typo: `cd build`

File: README.md [hunk 2]
```diff
@@ -71,3 +73,34 @@ Examples:
...
+cmake -GNinja
+cmake --build .
+```
+
+Build and run the unit tests
+----------------------------
+
+```
+mkdir build && cd make
```
Comment (jhasse) [inline README.md:93]: typo: `cd build`

**Review by jhasse — COMMENTED** [review]
Thanks! I found two typos otherwise I really like your changes.

File: README.md (hunk 1 above)
Comment (pierre-dejoue) [reply README.md:84]: thks!

File: poly2tri/common/shapes.cc [hunk 3]
```diff
@@ -243,15 +243,28 @@ Point* Triangle::PointCCW(const Point& point)
...
+}
+
 // The neighbor clockwise to given point
 Triangle* Triangle::NeighborCW(const Point& point)
 {
   if (&point == points_[0]) {
     return neighbors_[1];
   } else if (&point == points_[1]) {
     return neighbors_[2];
+  } else {
```
Comment (jhasse) [inline poly2tri/common/shapes.cc:265]: I don't like this change, see https://clang.llvm.org/extra/clang-tidy/checks/readability-else-after-return.html
Can you keep the return without else?

Comment (pierre-dejoue) [reply poly2tri/common/shapes.cc]: Sure, no problem. Thanks for the link!

**Comment by jhasse** [comment]
Thanks!

**Pull request merged**